### ISSUE TITLE
Git branch status color not working in tree-view with Seti UI Theme

### ISSUE THREAD

**Issue by gagregrog** [subesokun/atom-tree-view-git-status#56]
Hello! I love your package, it's hugely beneficial. 

The package works fine for me with the default Atom themes, but when I switch to the Seti UI theme the color status does not show in tree view. I don't know enough about Atom packages to say if this is an issue with the Seti UI package, or with your package.

Initially, the lack of color was the only issue I encountered. However, recently the branch name itself stopped showing in tree-view. I tried uninstalling and reinstalling these packages individually, but that did not fix the problem. Ultimately I completely uninstalled (including all hidden files/folders and preference files) and reinstalled Atom, then reinstalling each of my packages one by one. That seemed to fix the branch-name problem, but the color still does not reflect the branch status.

Any ideas on how I can fix this?

Thanks!

### Atom Dark UI:
<img width="177" alt="screen shot 2017-10-28 at 11 20 19 am" src="https://user-images.githubusercontent.com/20693762/32137309-396d4b32-bbd2-11e7-9074-cd2b694b01c7.png">

### Seti UI:
<img width="168" alt="screen shot 2017-10-28 at 11 20 47 am" src="https://user-images.githubusercontent.com/20693762/32137312-43d5702c-bbd2-11e7-9b2f-f93706663b2e.png">

---

Package Version: 1.4.0

OS: macOS High Sierra 10.13
Atom Version: 1.21.1
Atom UI Theme: Seti
Atom Syntax Theme: Seti

**Comment by gagregrog** [comment]
Update: 

After restarting the computer, git status is no longer showing again.

<img width="295" alt="screen shot 2017-10-28 at 1 13 06 pm" src="https://user-images.githubusercontent.com/20693762/32138090-fa9f5cd2-bbe1-11e7-8ef8-cad8d1c2aac4.png">

Even though `master` branch is indicated in the bottom right corner.

<img width="114" alt="screen shot 2017-10-28 at 1 13 24 pm" src="https://user-images.githubusercontent.com/20693762/32138094-0e5576b2-bbe2-11e7-9241-a4613d0692f0.png">

With each new package that I installed I reloaded the atom workspace using ctrl-opt-cmd-L, thinking that would suffice, but I guess the problem didn't present itself until the full restart of the system.

Not sure what to do other than to repeat the same process, but complete an actual reboot between each new package install. Any other ideas of something I might try?

Thank you!

**Comment by subesokun** [comment]
Thanks for using my package :) I can confirm the color issue but I need to investigate if that is a Seti UI issue or if I can do something about that. The issue with the missing branch name I can't reproduce so far.  Do you've some other package installed that has influence on the tree-view?

**Comment by gagregrog** [comment]
Hi @subesokun, thanks for responding so quickly!

I have quite a few packages that I use, but the only one that I think would act upon the tree-view would be `file-icons`.

Below is the complete list of community packages I have installed:

`atom-beautify` 0.30.5
`atom-ternjs` 0.18.3
`busy-signal` 1.4.3
`color-picker` 2.2.5
`file-icons` 2.1.11
`git-log` 0.4.1
`git-time-machine` 1.5.9
`highlight-selected` 0.13.1
`ide-typescript` 0.6.2
`intentions` 1.1.5
`language-typescript` 0.2.1
`linter` 2.2.0
`linter-eslint` 8.2.1
`linter-ui-default` 1.6.7
`merge-conflicts` 1.4.5
`minimap` 4.29.6
`minimap-pigments` 0.2.2
`open-in-browser` 0.5.2
`open-recent` 5.0.0
`pigments` 0.40.2
`script` 3.17.3
`sync-settings` 0.8.3
`todo-show` 2.1.0
`touchbar` 0.6.1
`tree-view-git-status` 1.4.0

Themes:
`monokai-dark` 1.3.0 (not in use)
`seti-syntax` 1.1.3 (in use)
`seti-ui` 1.9.0 (in use)

I've uninstalled everything again, so am going to start fresh and see if I have better luck pinpointing where it breaks this time.

Thanks!

**Comment by gagregrog** [comment]
Alright, well now I'm even more confused than ever. I completely uninstalled everything, and upon reinstalling Atom, tree-view-git-status no longer shows my branch name even when it is the first package I install.

I removed atom via the following Terminal commands, could that have anything to do with these issues?

```
rm -rf ~/.atom
rm -rf /usr/local/bin/atom
rm -rf /usr/local/bin/apm
rm -rf /Applications/Atom.app
rm -rf ~/Library/Preferences/com.github.atom.plist
rm -rf ~/"Library/Application Support/com.github.atom.ShipIt"
rm -rf ~/"Library/Application Support/Atom"
rm -rf ~/"Library/Saved Application State/com.github.atom.savedState"
rm -rf ~/Library/Caches/com.github.atom
rm -rf ~/Library/Caches/com.github.atom.Shipit
rm -rf ~/Library/Caches/Atom
```

**Comment by gagregrog** [comment]
I don't know if this will help, but I opened the dev tools and was given this warning. Do you think it is connected to the issue?

<img width="553" alt="screen shot 2017-10-28 at 2 55 11 pm" src="https://user-images.githubusercontent.com/20693762/32138891-2baa74ca-bbf0-11e7-9230-ed80e6ab8cac.png">

**Comment by gagregrog** [comment]
...and it's gotten even stranger. After working on a branch for about an hour or so, I noticed that the branch name magically had appeared. I don't know when, and I don't know why, but it was there.

After a restart it was gone again. Is there something that could be happening during the boot process that would stop it from working, that would somehow magically fix itself after a given amount of time?

I tried replicating specific things I had done during my session, but wasn't able to get it to show up again.

However, I just checked on a different computer that runs the same setup (atom-wise) and the branch label does show there, so that leads me to believe that it is in fact something on my other computer that's interfering with atom, rather than a package that's interfering with it.

Do you have any idea how I might go about troubleshooting that?

**Comment by subesokun** [comment]
Sorry to hear that you've that much troubles. Usually just disabling other packages is already enough and you don't have to reinstall Atom itself or remove other packages.

Regarding the disappearing branch name, well that happens if the tree-view refreshes but doesn't inform the tree-view-git-status package about it. A tree-view refresh can be triggered by many things (resizing window, change projects, ...) so it occurs also while Atom is running. So if the tree-view-git-status package doesn't get informed about this update the git status will suddenly disappear as on every refresh the complete DOM tree of the tree-view gets reset. So far I've no clue yet what in your case goes wrong that the refresh event gets lost.

The screenshot from above, seems that is some tree-view internal issue but might not be related to your problem, see https://discuss.atom.io/t/answered-warning-intersectionobserver-observe-target/46650

In your listed packages the only interesting package would be [ide-typescript](https://atom.io/packages/ide-typescript) as it might affect the tree-view. But I'm wondering if the package really works as expected as you don't have the basic [Atom IDE UI](https://atom.io/packages/atom-ide-ui) installed. I've also installed Atom IDE UI and had no issues so far though.

**Comment by subesokun** [comment]
btw. if you like dark themes you could try using the pre-installed `One Dark` theme and see if you've there the same issues. If not, then the disappearing branch name issue might be really related to the Seti UI theme.

**Comment by gagregrog** [comment]
Thanks for the info.

There are some additional peculiarities. For example, I noticed that after working on a branch for about an hour or so, that the branch status at some point had magically reappeared in tree view.

I then restarted my computer to see if that would affect it, and sure enough after restarting the branch status was gone again. I tried replicating everything I did during my work in an attempt to get it to work again, but had no luck.

On top of that, I loaded all of the same Atom packages and settings (using the sync-settings package) on a different machine, and the branch status worked fine. I then ended up loading the same settings and packages on a third computer, and again the branch status appeared without issue. However, after updating some software and installing a few other applications the branch status stopped working.

So, it would seem that there is something on my system related to either those last updates (most recent iTunes update and xCode update) or the apps I downloaded (tuck, cinch, and size-up, all from irradiated software) that is creating the issue.

I'm going to continue to troubleshoot the issue on my end, but it doesn't appear to have anything to do with your wonderful package, so I'm going to go ahead and close this issue.

Thanks for your help in troubleshooting! If I figure anything out I'll report back.

Cheers,
Rob
